pkg/controllers/user/serviceaccount/serviceaccount.go

```diff
@@ -0,0 +1,271 @@
+package serviceaccount
+
+import (
+	"fmt"
+	"strings"
+
+	policytemplate "github.com/rancher/rancher/pkg/controllers/user/template"
+	v13 "github.com/rancher/types/apis/core/v1"
+	"github.com/rancher/types/apis/extensions/v1beta1"
+	"github.com/rancher/types/apis/management.cattle.io/v3"
+	v12 "github.com/rancher/types/apis/rbac.authorization.k8s.io/v1"
+	"github.com/rancher/types/config"
+	"github.com/sirupsen/logrus"
+	"k8s.io/api/core/v1"
+	v1beta12 "k8s.io/api/extensions/v1beta1"
+	rbac "k8s.io/api/rbac/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/labels"
+)
+
+const (
+	apiGroup                     = "rbac.authorization.k8s.io"
+	apiVersion                   = "extensions/v1beta1"
+	podSecurityAnnotation        = "serviceaccount.cluster.cattle.io/pod-security"
+	podSecurityVersionAnnotation = "serviceaccount.cluster.cattle.io/pod-security-version"
+	projectIdAnnotation          = "field.cattle.io/projectId"
+)
+
+func Register(context *config.UserContext) {
+	logrus.Infof("registering serviceaccount handler for cluster %v", context.ClusterName)
+
+	m := &manager{
+		clusters: context.Management.Management.Clusters(""),
+		bindings: context.RBAC.RoleBindings(""),
+		policies: context.Extensions.PodSecurityPolicies(""),
+		roles:    context.RBAC.Roles(""),
+
+		clusterLister:   context.Management.Management.Clusters("").Controller().Lister(),
+		templateLister:  context.Management.Management.PodSecurityPolicyTemplates("").Controller().Lister(),
+		policyLister:    context.Extensions.PodSecurityPolicies("").Controller().Lister(),
+		bindingLister:   context.RBAC.RoleBindings("").Controller().Lister(),
+		roleLister:      context.RBAC.Roles("").Controller().Lister(),
```

Comment:
We should be using ClusterRole not Role for create the role. This will allow to create fewer objects in etcd because we can reuse the ClusterRole across all the appropriate namesapces. With Role, we would have to create one per namespace.

Note that when you switch to using ClusterRole, the binding should still be the namepsaced RoleBinding, not the cluster-wide ClusterRoleBinding.